onsstove/technology.py

```diff
@@ -86,206 +86,74 @@ def relative_risk(self):
-        End year of the analysis
-    arg3: tech
-        Stove type assessed
-    arg4: discount_rate
-        Discount rate to extrapolate costs
-    arg5: hhsize_R
-        Rural household size
-    arg6: hhsize_U
-        Urban household size
-    arg7: sfu
-        Solid fuel users (ration)
-
-    Returns
-    ----------
-    Monetary morbidity for each stove in urban and rural settings
-    """
-
-    if tech.pm25 < 7.298:
-        rr_alri = 1
-    else:
-        rr_alri = 1 + 2.383 * (1 - exp(-0.004 * (tech.pm25 - 7.298) ** 1.193))
-
-    if tech.pm25 < 7.337:
-        rr_copd = 1
-    else:
-        rr_copd = 1 + 22.485 * (1 - exp(-0.001 * (tech.pm25 - 7.337) ** 0.694))
-
-    if tech.pm25 < 7.505:
-        rr_ihd = 1
-    else:
-        rr_ihd = 1 + 2.538 * (1 - exp(-0.081 * (tech.pm25 - 7.505) ** 0.466))
-
-    if tech.pm25 < 7.345:
-        rr_lc = 1
-    else:
-        rr_lc = 1 + 152.496 * (1 - exp(-0.000167 * (tech.pm25 - 7.345) ** 0.76))
-
-    paf_alri = (sfu * (rr_alri - 1)) / (sfu * (rr_alri - 1) + 1)  # sfu is one.
-    paf_copd = (sfu * (rr_copd - 1)) / (sfu * (rr_copd - 1) + 1)
-    paf_ihd = (sfu * (rr_ihd - 1)) / (sfu * (rr_ihd - 1) + 1)
-    paf_lc = (sfu * (rr_lc - 1)) / (sfu * (rr_lc - 1) + 1)
-
-    morb_alri_U = hhsize_U * paf_alri * inci_alri
-    morb_copd_U = hhsize_U * paf_copd * inci_copd
-    morb_ihd_U = hhsize_U * paf_ihd * inci_ihd
-    morb_lc_U = hhsize_U * paf_lc * inci_lc
 
-    morb_alri_R = hhsize_R * paf_alri * inci_alri
-    morb_copd_R = hhsize_R * paf_copd * inci_copd
-    morb_ihd_R = hhsize_R * paf_ihd * inci_ihd
-    morb_lc_R = hhsize_R * paf_lc * inci_lc
+    def morbidity(self, social_specs_file, paf_0_alri, paf_0_copd, paf_0_lc, paf_0_ihd):
+        """
+        Calculates morbidity rate per fuel
 
-    cl_copd = {1: 0.3, 2: 0.2, 3: 0.17, 4: 0.17, 5: 0.16}
-    cl_alri = {1: 0.7, 2: 0.1, 3: 0.07, 4: 0.07, 5: 0.06}
-    cl_lc = {1: 0.2, 2: 0.1, 3: 0.24, 4: 0.23, 5: 0.23}
-    cl_ihd = {1: 0.2, 2: 0.1, 3: 0.24, 4: 0.23, 5: 0.23}
+        Returns
+        ----------
+        Monetary morbidity for each stove in urban and rural settings
+        """
 
-    i = 1
-    morb_U_vector = []
-    morb_R_vector = []
-    while i < 6:
-        morbidity_alri_U = cl_alri[i] * coi_alri * morb_alri_U / (1 + discount_rate_social) ** (end_year - start_year)
-        morbidity_copd_U = cl_copd[i] * coi_copd * morb_copd_U / (1 + discount_rate_social) ** (end_year - start_year)
-        morbidity_lc_U = cl_lc[i] * coi_lc * morb_lc_U / (1 + discount_rate_social) ** (end_year - start_year)
-        morbidity_ihd_U = cl_ihd[i] * coi_ihd * morb_ihd_U / (1 + discount_rate_social) ** (end_year - start_year)
+        rr_alri, rr_copd, rr_ihd, rr_lc = relative_risk(self)
 
-        morb_U_total = morbidity_alri_U + morbidity_copd_U + morbidity_lc_U + morbidity_ihd_U
+        paf_alri = paf(rr_alri, sfu)
+        paf_copd = paf(rr_copd, sfu)
+        paf_ihd = paf(rr_ihd, sfu)
+        paf_lc = paf(rr_lc, sfu)
```

Comment:
As `paf` is now a class method, then it also needs to be called as `self.paf(agr1, agr2)`

Reply:
Fixed